metrics/influx.go

```diff
@@ -0,0 +1,165 @@
+package metrics
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/prebid/prebid-cache/config"
+	"github.com/rcrowley/go-metrics"
+	"github.com/vrischmann/go-metrics-influxdb"
+)
+
+type MetricsEntry struct {
+	Duration   metrics.Timer
+	Errors     metrics.Meter
+	BadRequest metrics.Meter
+	Request    metrics.Meter
+}
+
+type MetricsEntryByFormat struct {
+	Duration       metrics.Timer
+	Errors         metrics.Meter
+	BadRequest     metrics.Meter
+	JsonRequest    metrics.Meter
+	XmlRequest     metrics.Meter
+	DefinesTTL     metrics.Meter
+	InvalidRequest metrics.Meter
+	RequestLength  metrics.Histogram
+}
+
+type ConnectionMetrics struct {
```

Comment:
For Prometheus, I highly recommend to record two counters, one for opened and one for closed metrics. This is what we do in Prebid Server.

Reply:
Got it, do you also want to add connections closed, as we do in prebid-server?
```
 54 type PrometheusConnectionMetrics struct {
 55     - ActiveConnections      prometheus.Counter
 56     - ConnectionCloseErrors  prometheus.Counter
 57     - ConnectionAcceptErrors prometheus.Counter
 58
 59     + ConnectionsOpened   prometheus.Counter
 60     + ConnectionsErrors   *prometheus.CounterVec // the "Connection_error" label will hold the values "accept" or "close"
 61 }
metrics/prometheus.go
```

Reply:
Yeah. You really just need connections opened and closed. Active can be inferred.